rewrite-core/src/main/java/org/openrewrite/trait/Reference.java

```diff
@@ -31,16 +31,16 @@ enum Kind {
 
     String getValue();
 
-    default boolean supportsRename() {
-        return false;
-    }
-
     default boolean matches(Matcher matcher) {
         return matcher.matchesReference(this);
     }
 
-    default TreeVisitor<Tree, ExecutionContext> rename(Renamer renamer, String replacement) {
-        return renamer.rename(replacement);
+    default boolean supportsRename() {
+        return false;
+    }
+
+    default Tree rename(Renamer rename, Cursor cursor, ExecutionContext ctx) {
+        return cursor.getValue();
```

Comment:
Unlike the previous version, the variants will never work unless it's overridden. I think the default implementation should just contain throw a UnsupportedOperationException.

Reply:
My thoughts exactly